### PR TITLE
X509 Credentials Provider Binding

### PR DESCRIPTION
**Pull request by bretambrose**
https://github.com/aws/aws-iot-device-sdk-cpp-v2/issues/129

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: include/aws/crt/auth/Credentials.h [hunk 1]
```diff
@@ -6,6 +6,8 @@
 
 #include <aws/crt/Exports.h>
 #include <aws/crt/Types.h>
+#include <aws/crt/http/HttpConnection.h>
```
Comment (JonathanHenson) [inline include/aws/crt/auth/Credentials.h:9]: I don’t see where this include is being used.

File: source/auth/Credentials.cpp [hunk 2]
```diff
@@ -267,6 +269,45 @@ namespace Aws
...
+                AWS_ZERO_STRUCT(proxy_options);
+                if (config.ProxyOptions.has_value())
+                {
+                    const Http::HttpClientConnectionProxyOptions &proxy_config = config.ProxyOptions.value();
+
+                    proxy_options.host = aws_byte_cursor_from_c_str(proxy_config.HostName.c_str());
+                    proxy_options.port = proxy_config.Port;
+                    proxy_options.tls_options = proxy_config.TlsOptions->GetUnderlyingHandle();
+                    proxy_options.auth_type = (enum aws_http_proxy_authentication_type)proxy_config.AuthType;
+                    proxy_options.auth_username = aws_byte_cursor_from_c_str(proxy_config.BasicAuthUsername.c_str());
```
Comment (JonathanHenson) [inline source/auth/Credentials.cpp:296]: Some of these are optional right? Should we do a check that they’ve been set?

File: include/aws/crt/auth/Credentials.h (hunk 1 above)
Comment (bretambrose) [reply include/aws/crt/auth/Credentials.h:9]: Proxy Options is there

File: source/auth/Credentials.cpp (hunk 2 above)
Comment (bretambrose) [reply source/auth/Credentials.cpp:296]: username and password being null will just result in 0-length cursors.  Everything else is required.

**Pull request merged**